File: Data Structures & Algorithms/binary-search/submission-0.java
class Solution {
    public int search(int[] nums, int target) {
        int L = 0;
        int R = nums.length - 1;
        
        while(L <= R) {
            int index = (R + L) / 2;
            if (nums[index] == target) {
                return index;
            } else if (nums[index] > target){
                R = index - 1;
            } else {
                L = index + 1;
            }
        }

        return -1;
    }
}

File: Data Structures & Algorithms/products-of-array-discluding-self/submission-0.py
class Solution:
    def productExceptSelf(self, nums: List[int]) -> List[int]:
        ans = []
        n = len(nums)

        for i in range(n):
            total = 1

            for j in range(n):
                if i == j:
                    continue
                total *= nums[j]

            ans.append(total)

        return ans

